app/javascript/packs/bulk_operations.ts

```diff
@@ -0,0 +1,152 @@
+  const dataModelName = 'data-association-name'
+  const dataDefaultText = 'data-default-text'
+
+  const colorboxOpts = {
+    autoDimensions: true,
+    overlayShow: true, // cannot use modal, because its setting cannot be overridden
+    hideOnOverlayClick: false,
+    hideOnContentClick: false,
+    enableEscapeButton: false,
+    showCloseButton: true
+  }
+
+  function hrefFor (url: string) {
+    // url address might already include some parameters
+    const connector = url.includes('?') ? '&' : '?'
+    let href = url.concat(connector, $('table tbody .select :checked').serialize())
+
+    const selectTotalEntries = document.querySelector<HTMLAnchorElement>('#bulk-operations a.select-total-entries')
+    if (selectTotalEntries?.hasAttribute(dataSelectedTotal)) {
+      href += '&selected_total_entries=true'
+    }
+
+    return href
+  }
+
+  function prepareSelectAllCheckbox () {
+    document.querySelector<HTMLInputElement>('table thead .select .select-all')!
+      .addEventListener('change', (event) => {
+        const checked = (event.target as HTMLInputElement).checked
+
+        document.querySelectorAll<HTMLTableRowElement>('table tbody tr')
+          .forEach((row) => {
+            row.classList.toggle('selected', checked)
+
+            const checkbox = row.querySelector<HTMLInputElement>('.select input[type=checkbox]')!
+            checkbox.checked = checked
+            checkbox.setAttribute('checked', checked.toString())
+          })
+
+        updateBulkOperationsCard()
+      })
+  }
+
+  function prepareSingleCheckboxes () {
+    document.querySelectorAll<HTMLInputElement>('table tbody .select input[type=checkbox]')
+      .forEach(checkbox => {
+        checkbox.addEventListener('change', () => {
+          const row = checkbox.closest('tr')!
+          row.classList.toggle('selected', checkbox.checked)
+
+          updateBulkOperationsCard()
+          updateSelectAllCheckbox()
+        })
+      })
+  }
+
+  function updateSelectAllCheckbox () {
+    const selected = selectedRows()
+    const unselected = document.querySelectorAll('table tbody tr:not(.selected').length
+
+    const checkbox = document.querySelector<HTMLInputElement>('table thead .select .select-all')!
+
+    checkbox.indeterminate = selected > 0 && unselected > 0
+    checkbox.checked = unselected === 0
+  }
+
+  function setSelectedCount (count: number | string) {
+    document.querySelector<HTMLSpanElement>('#bulk-operations .count')!.innerText = count.toString()
+  }
+
+  function updateBulkOperationsCard () {
+    const bulk = $('#bulk-operations')
+    const selected = selectedRows()
+
+    if (selected > 0) {
+      bulk.slideDown()
+      setSelectedCount(selected)
+
+      const selectTotalEntries = document.querySelector<HTMLAnchorElement>('#bulk-operations a.select-total-entries')
+      if (selectTotalEntries) {
+        const isAllSelected = document.querySelectorAll('table tbody tr:not(.selected').length === 0
```

Comment:
```suggestion
        const isAllSelected = document.querySelectorAll('table tbody tr:not(.selected)').length === 0
```
?

Reply:
Actually, there are some queries that I think might be useful to extract to functions, so they are not repeated.

This is one of them, and another one is:
```
document.querySelector<HTMLAnchorElement>('#bulk-operations a.select-total-entries')
```
which repeats 3 times.

Reply:
Done

Reply:
Done